File: CS/ExecuteSprocWorkflowExample.Module/CustomNativeActivity.cs
using DevExpress.ExpressApp;
using DevExpress.ExpressApp.Xpo;
using DevExpress.Workflow.Activities;
using System;
using System.Activities;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ExecuteSprocWorkflowExample.Module {
    public class CustomNativeActivity : NativeActivity {
        protected virtual IObjectSpace GetObjectSpace(NativeActivityContext context) {
            return (IObjectSpace)context.Properties.Find(ObjectSpaceTransactionScope.ObjectSpacePropertyName);
        }
        protected override void Execute(NativeActivityContext context) {
            XPObjectSpace xpObjectSpace = (XPObjectSpace)GetObjectSpace(context);
            xpObjectSpace.Session.ExecuteSproc("StoredProcName");
        }
    }
}

File: VB/ExecuteSprocWorkflowExample.Module/CustomNativeActivity.vb
Imports DevExpress.ExpressApp
Imports DevExpress.ExpressApp.Xpo
Imports DevExpress.Workflow.Activities
Imports System
Imports System.Activities
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text
Imports System.Threading.Tasks

Namespace ExecuteSprocWorkflowExample.Module
    Public Class CustomNativeActivity
        Inherits NativeActivity

        Protected Overridable Function GetObjectSpace(ByVal context As NativeActivityContext) As IObjectSpace
            Return DirectCast(context.Properties.Find(ObjectSpaceTransactionScope.ObjectSpacePropertyName), IObjectSpace)
        End Function
        Protected Overrides Sub Execute(ByVal context As NativeActivityContext)
            Dim xpObjectSpace As XPObjectSpace = DirectCast(GetObjectSpace(context), XPObjectSpace)
            xpObjectSpace.Session.ExecuteSproc("StoredProcName")
        End Sub
    End Class
End Namespace
